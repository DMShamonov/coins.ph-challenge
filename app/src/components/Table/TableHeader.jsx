// @flow

import * as React from 'react';
import classnames from 'classnames';

import type { TableHeaderPropsType } from './types.js.flow';

export default function TableHeader({
  className,
  children,
}: TableHeaderPropsType): React.Node {
  return (
    <div className={classnames('table__header', className)}>
      {children}
    </div>
  );
}
