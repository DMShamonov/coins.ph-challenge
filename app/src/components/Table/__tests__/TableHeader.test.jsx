import React from 'react';
import { shallow } from 'enzyme';

import TableHeader from '../TableHeader';

describe('`TableHeader` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow(<TableHeader>__CONTENT__</TableHeader>)).toMatchSnapshot();
  });

  test('with `className` prop', () => {
    expect(shallow(<TableHeader className="__CLASSNAME__">__CONTENT__</TableHeader>)).toMatchSnapshot();
  });
});
