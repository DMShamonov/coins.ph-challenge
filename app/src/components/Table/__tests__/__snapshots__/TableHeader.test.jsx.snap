// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`TableHeader\` component should render correctly with \`className\` prop 1`] = `
<div
  className="table__header __CLASSNAME__"
>
  __CONTENT__
</div>
`;

exports[`\`TableHeader\` component should render correctly with default props 1`] = `
<div
  className="table__header"
>
  __CONTENT__
</div>
`;
